src/modules/previewpane/MarkDownPreviewHandler/HTMLParsingExtension.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Markdig;
+using Markdig.Extensions.Figures;
+using Markdig.Extensions.Tables;
+using Markdig.Renderers;
+using Markdig.Renderers.Html;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+
+namespace MarkDownPreviewHandler
+{
+    /// <summary>
+    /// Markdig Extension to process html nodes in markdown AST.
+    /// </summary>
+    internal class HTMLParsingExtension : IMarkdownExtension
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HTMLParsingExtension"/> class.
+        /// </summary>
+        public HTMLParsingExtension(string baseUrl = "")
+        {
+            this.BaseUrl = baseUrl;
+        }
+
+        /// <summary>
+        /// Gets or sets path to directory containing markdown file.
+        /// </summary>
+        public string BaseUrl { get; set; }
+
+        /// <inheritdoc/>
+        public void Setup(MarkdownPipelineBuilder pipeline)
+        {
+            // Make sure we don't have a delegate twice
+            pipeline.DocumentProcessed -= this.PipelineOnDocumentProcessed;
+            pipeline.DocumentProcessed += this.PipelineOnDocumentProcessed;
+        }
+
+        /// <inheritdoc/>
+        public void Setup(MarkdownPipeline pipeline, IMarkdownRenderer renderer)

```

Comment:
Should this version of 'Setup' initialize the delegate as well?

Reply:
I am not implementing this version of initialization for the Extension. This is used if we first parse markdown to AST and then manually apply the extension. But since this is part of interface I had to declare function. Should I make it throw `notImlementedException` ?

Reply:
`Setup(MarkdownPipeline pipeline, IMarkdownRenderer renderer)` is called from inside renderers. But we are not using any renderer in this parsing context. The first setup function is called while [building pipeline](https://github.com/lunet-io/markdig/blob/master/src/Markdig/MarkdownPipelineBuilder.cs)

Reply:
Is this actually called, but from our perspective is a 'no-op', or is it not called.   If it's the former, you might want to add a comment about why we are implementing this interface, but don't have to provide an implementation for this override of 'Setup'.  If it's the latter you might want to throw a NotImplemented exception, so that it is will explicitly fail if the call patterns of markdig change.

Reply:
It is first case i.e It is called but from our perspective is a 'no-op' because it is independent of renderer context.